### PR TITLE
Cache Global Latest offset

### PR DESCRIPTION
**Pull request by adam-yorwerth**

### CODE REVIEW

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/OffsetFetcher.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class OffsetFetcher {
+
+    private static final PipeLogger LOG = new PipeLogger(LoggerFactory.getLogger(PostgresqlStorage.class));
+
+    private static final String currentTimestamp = "CURRENT_TIMESTAMP";
+
+    private final int readDelaySeconds;
+
+    public OffsetFetcher(
+        @Value("${persistence.read.read-delay-seconds:0}") final int readDelaySeconds
```
Comment (adam-yorwerth) [inline pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/OffsetFetcher.java:24]: Might not be a change in this PR, but I feel the default shouldn't be 0 here - can we make the default something which is safe?

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/OffsetFetcher.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    private String getSelectLatestOffsetQuery() {
+        String filterCondition = "WHERE created_utc >= %s - INTERVAL '%s SECONDS'";
+        return
+            "SELECT coalesce (" +
+                " (SELECT min(msg_offset) - 1 from events where msg_offset in " +
+                "   (SELECT msg_offset FROM events " + String.format(filterCondition, currentTimestamp, readDelaySeconds) + ")" +
+                " ), " +
+                " (SELECT max(msg_offset) FROM events), " +
+                " 0 " +
+                ") as last_offset;";
```
Comment (adam-yorwerth) [inline pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/OffsetFetcher.java:61]: Formatting of sql query could be neater

**Review by adam-yorwerth — COMMENTED** [review]
Approved

**Review by sebastianlavender — DISMISSED** [review]
LGTM - but probably need to rename the PR though

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/OffsetFetcher.java (hunk 1 above)
Comment (LorenzoCianciaruso) [reply pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/OffsetFetcher.java:24]: Good point.. I think it should be higher than what we currently have in the ymls. at the moment we have 60 in the ymls, so maybe 120?

Comment (ashish-sharma09) [reply pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/OffsetFetcher.java:24]: I'm not entirely sure even if we need default here. Default is just a lazy way of saying if it is not configured do not fail but use the default. Some important values should exist explicitly within the configuration.

**Review by sebastianlavender — APPROVED** [review]

**Pull request merged**